lib/Wasmtime.hs

```diff
@@ -1235,22 +1243,39 @@ kindToHaskellTypeStr = \case
 
 -- | Class of Haskell types that WASM @'Func'tions@ can take as parameters or
 -- return as results or which can be retrieved from and set to @'Global's@.
-class HasValType a where
+class Val a where
   kind :: Proxy a -> C'wasm_valkind_t
 
-instance HasValType Int32 where kind _proxy = c'WASMTIME_I32
+  pokeVal :: Ptr C'wasmtime_val_t -> a -> IO ()
+  default pokeVal :: (Storable a) => Ptr C'wasmtime_val_t -> a -> IO ()
```

Comment:
what is 'default'? we can also discuss this over lunch.

Reply:
I'm using [DefaultSignatures](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/default_signatures.html#extension-DefaultSignatures) here to give a default implementation in case the type has an instance for `Storable`.

Reply:
cool! I thought it's along these lines, but didnt realize that the default you are giving is for storable instances. very nice feature.